src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/RefAPISource.java

```diff
@@ -0,0 +1,122 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.api.services.genomics.Genomics;
+import com.google.api.services.genomics.model.ListBasesResponse;
+import com.google.api.services.genomics.model.Reference;
+import com.google.api.services.genomics.model.ReferenceSet;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.genomics.dataflow.utils.GCSOptions;
+import com.google.cloud.genomics.dataflow.utils.GenomicsOptions;
+import com.google.cloud.genomics.utils.GenomicsFactory;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.security.GeneralSecurityException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * RefAPISource makes calls to the Google Genomics API to get ReferenceBases. This is designed to used with workers.
+ * The service is created on each worker. We've benchmarked this and found that class instantiation is much cheaper
+ * than the API call (~1ms vs ~100ms).
+ */
+public class RefAPISource implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private transient Genomics genomicsService;
+
+    public RefAPISource() { }
+
+    /**
+     * Query the Google Genomics API for reference bases spanning the specified interval from the specified
+     * reference name.
+     *
+     * @param pipelineOptions -- are used to get the credentials necessary to call the Genomics API
+     * @param apiData - contains the hashmap that maps from reference name to Id needed for the API call.
+     * @param interval - the range of bases to retrieve.
+     * @return the reference bases specified by interval and apiData (using the Google Genomics API).
+     */
+    public ReferenceBases getReferenceBases(final PipelineOptions pipelineOptions, final RefAPIMetadata apiData, final SimpleInterval interval) {
+        Utils.nonNull(pipelineOptions);
+        Utils.nonNull(apiData);
+        Utils.nonNull(interval);
+        
+        if (genomicsService == null) {
+            genomicsService = createGenomicsService(pipelineOptions);
+        }
+        if ( !apiData.getReferenceNameToIdTable().containsKey(interval.getContig()) ) {
+            throw new UserException("Contig " + interval.getContig() + " not in our set of reference names for this reference source");
+        }
+
+        try {
+            final Genomics.References.Bases.List listRequest = genomicsService.references().bases().list(apiData.getReferenceNameToIdTable().get(interval.getContig()));
+            // We're subtracting 1 with the start but not the end because GA4GH is zero-based (inclusive,exclusive)
```

Comment:
Yes indeed. How about centralizing the conversion code in SimpleInterval? We could have getGA4GHStart(), getGA4GHEnd().